src/components/common/CodeEditor.js

```diff
@@ -0,0 +1,289 @@
+        variant="outlined"
+        label={t('Commit Message')}
+        placeholder={t('Updated code sample')}
+        onChange={this.handleChangeTextField('commitMessage')}
+        value={commitMessage}
+        fullWidth
+        helperText={commitMessageTooLong ? t(COMMIT_MESSAGE_TOO_LONG) : ''}
+        error={commitMessageTooLong}
+      />
+    );
+
+    const extendedCommitDescriptionControl = (
+      <TextField
+        size="small"
+        color="primary"
+        variant="outlined"
+        label={t('Optional Extended Description')}
+        placeholder={t('Add an optional description')}
+        onChange={this.handleChangeTextField('commitDescription')}
+        value={commitDescription}
+        fullWidth
+        multiline
+        maxRows={3}
+      />
+    );
+
+    return (
+      <>
+        <Grid
+          container
+          className={classes.container}
+          direction="column"
+          spacing={2}
+        >
+          <Paper variant="outlined" className={classes.paper}>
+            <Grid className={classes.editor} item>
+              <Editor
+                height="40vh"
+                defaultLanguage={programmingLanguage}
+                language={programmingLanguage}
+                value={code}
+                onChange={this.handleChangeEditor('code')}
+                options={{
+                  scrollBeyondLastLine: false,
+                  detectIndentation: false,
+                  tabSize: 2,
+                }}
+              />
+            </Grid>
+            <Divider className={classes.divider} />
+            <Grid
+              container
+              item
+              direction="column"
+              alignContent="stretch"
+              spacing={1}
+            >
+              <Grid item className={classes.commitTextField}>
+                {commitMessageControl}
+              </Grid>
+              <Grid item className={classes.commitTextField}>
+                {extendedCommitDescriptionControl}
+              </Grid>
+              <Grid container item direction="row">
+                <Grid item>
+                  <Button
+                    variant="contained"
+                    color="primary"
+                    className={classes.button}
+                    onClick={this.handleCommit}
+                    disabled={!hasChanged}
+                  >
+                    {t('Commit changes')}
+                  </Button>
+                </Grid>
+                <Grid item>
+                  <Button
+                    variant="contained"
+                    color="secondary"
+                    className={classes.button}
+                    onClick={this.handleClose}
+                  >
+                    {t('Cancel')}
+                  </Button>
+                </Grid>
+              </Grid>
+            </Grid>
+          </Paper>
+        </Grid>
+      </>
+    );
+  };
+
+  render() {
+    return this.renderCodeEditor();
+  }
+}
```

Comment:
This should probably include the code in the helper function.

Reply:
I removed the helper function and put the code in the `render` function instead. was this what you meant ?